src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -40,14 +40,20 @@ public WebHostBuilder()
             _configureServicesDelegates = new List<Action<IServiceCollection>>();
             _configureLoggingDelegates = new List<Action<ILoggerFactory>>();
 
+            _config = new ConfigurationBuilder()
+                .AddEnvironmentVariables(prefix: "ASPNETCORE_")
+                .Build();
+
             // This may end up storing null, but that's indistinguishable from not adding it.
             UseSetting(WebHostDefaults.EnvironmentKey, Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT")
```

Comment:
Interesting C# quirk. You cannot call an extension method of the interface from the class that is implementing the interface unless it's done explicitly. This means that `UseConfiguration` which is an extension on `IWebHostBuilder` cannot be called from `WebHostBuilder` unless you call it like `HostingAbstractionsWebHostBuilderExtensions.UseConfiguration(this, config)`. Hmm, since that looks ugly as well, I suggest we just keep the current implementation.